nixops/backends/__init__.py

```diff
@@ -12,14 +12,12 @@
 class MachineDefinition(nixops.resources.ResourceDefinition):
     """Base class for NixOps machine definitions."""
 
-    def __init__(self, xml, config={}):
+    def __init__(self, xml, config={}) -> None:
         nixops.resources.ResourceDefinition.__init__(self, xml, config)
-        self.encrypted_links_to = set(
-            [
-                e.get("value")
-                for e in xml.findall("attrs/attr[@name='encryptedLinksTo']/list/string")
-            ]
-        )
+        self.encrypted_links_to = {
```

Comment:
Hmm not sure about changing this to a dict from a set?

Reply:
Old way was list comprehension passed to set (which unnecessarily builds a list and then throws it away), edited way is set comprehension (syntax new in Python 2.7, see end of sets section in the official tutorial: https://docs.python.org/3/tutorial/datastructures.html#sets).